tests/integration/components/o-s-s/progress-bar-test.ts

```diff
@@ -0,0 +1,95 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | o-s-s/progress-bar', function (hooks) {
+  setupRenderingTest(hooks);
+  hooks.beforeEach(function () {
+    this.checkedValue = 42;
+  });
+
+  module('Successful render', function () {
+    test('it renders', async function (assert) {
+      await render(hbs`<OSS::ProgressBar @value={{this.checkedValue}} />`);
+
+      assert.dom('.oss-progress-bar').exists();
+    });
+
+    test('it renders only the progressBar when only @value is specified', async function (assert) {
+      await render(hbs`<OSS::ProgressBar @value={{this.checkedValue}} />`);
+
+      assert.dom('.oss-progress-bar').exists();
+      assert.dom('.oss-progress-bar--label').doesNotExist();
+      assert.dom('.oss-progress-bar--value').doesNotExist();
+    });
+
+    test('It renders the progress bar inner when the @value is specified', async function (assert) {
+      await render(hbs`<OSS::ProgressBar @value={{this.checkedValue}} />`);
+
+      assert.dom('.oss-progress-bar').exists();
+      const innerBar = this.element.querySelector('.oss-progress-bar--inner');
+      assert.dom(innerBar).exists();
+    });
+  });
+
+  module('@value behaviour', function () {
+    test('It renders the progress bar inner with the correct width when @value is specified', async function (assert) {
+      await render(hbs`<OSS::ProgressBar @value={{this.checkedValue}} />`);
+
+      assert.dom('.oss-progress-bar').exists();
+      const innerBar = this.element.querySelector('.oss-progress-bar--inner');
+      if (innerBar) {
```

Comment:
Why using a `if` here? Same below

Reply:
Didn't like the `Element | null `type in the assert , now I force it to be an `Element` only . If the element is null then the test fails anyway